qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/customers/CustomerDbConnector.groovy

```diff
@@ -81,23 +152,122 @@ class CustomerDbConnector implements CreateCustomerDataSource, UpdateCustomerDat
   @Override
   void addCustomer(Customer customer) throws DatabaseQueryException {
     try {
-      databaseQueries.addCustomer(customer)
-    } catch (DatabaseQueryException e) {
-      throw new DatabaseQueryException(e.message)
+      if (customerExists(customer)) {
+        throw new DatabaseQueryException("Customer is already in the database.")
+      }
+      Connection connection = connectionProvider.connect()
+      connection.setAutoCommit(false)
+
+      connection.withCloseable {it ->
+        try {
+          int customerId = createNewCustomer(it, customer)
+          storeAffiliation(it, customerId, customer.affiliations)
+          connection.commit()
+        } catch (Exception e) {
+          log.error(e.message)
+          log.error(e.stackTrace.join("\n"))
+          connection.rollback()
+          connection.close()
+          throw new DatabaseQueryException("Could not create customer.")
+        }
+      }
+    } catch (DatabaseQueryException ignored) {
+      throw new DatabaseQueryException("The customer could not be created: ${customer.toString()}")
```

Comment:
Not needed, as the use case knows about the Customer it wanted to create and can handle this information.